ComfortZoneApp/app/src/main/java/com/example/comfortzone/ui/HostActivity.java

```diff
@@ -88,14 +96,22 @@ private void requestPermissions() {
     }
 
     private void maybeUpdateComfortLevel() {
-        ParseUser currentUser = ParseUser.getCurrentUser();
+        currentUser = ParseUser.getCurrentUser();
         ArrayList<ComfortLevelEntry> todayEntries = (ArrayList<ComfortLevelEntry>) currentUser.get(ComfortLevelUtil.KEY_TODAY_ENTRIES);
         if (!todayEntries.isEmpty() && todayEntries.get(0).getUpdatedAt() != null && !DateUtils.isToday(todayEntries.get(0).getUpdatedAt().getTime())) {
             ComfortLevelUtil.updateComfortLevel(currentUser);
             ComfortCalcUtil.calculateAverages(currentUser);
         }
     }
 
+    private void getObjects() {
+        savedCities = new HashSet<Integer>();
+        ArrayList<Integer> savedIds = (ArrayList<Integer>) currentUser.get(KEY_SAVED_CITIES);
```

Comment:
I've been wondering, does this not need to run in background thread? since you are doing network call?